site/pages/create-disruption.page.tsx

```diff
@@ -1,28 +1,144 @@
 import { NextPageContext } from "next";
 import { parseCookies } from "nookies";
-import { ReactElement, useState } from "react";
+import { Fragment, ReactElement, SyntheticEvent, useState } from "react";
+import { z } from "zod";
 import ErrorSummary from "../components/ErrorSummary";
 import Checkbox from "../components/form/Checkbox";
 import DateSelector from "../components/form/DateSelector";
 import Radios from "../components/form/Radios";
 import Select from "../components/form/Select";
+import Table from "../components/form/Table";
 import TextInput from "../components/form/TextInput";
 import TimeSelector from "../components/form/TimeSelector";
 import { BaseLayout } from "../components/layout/Layout";
 import { DISRUPTION_REASONS, COOKIES_DISRUPTION_INFO, COOKIES_DISRUPTION_ERRORS } from "../constants/index";
-import { PageState } from "../interfaces";
-import { createDisruptionSchema, Disruption } from "../schemas/create-disruption.schema";
-import { getStateUpdater } from "../utils/formUtils";
+import { ErrorInfo, PageState } from "../interfaces";
+import {
+    createDisruptionSchema,
+    validitySchemaRefined,
+    Validity,
+    validitySchema,
+} from "../schemas/create-disruption.schema";
+import { flattenZodErrors } from "../utils";
 
 const title = "Create Disruptions";
 const description = "Create Disruptions page for the Create Transport Disruptions Service";
 
-export interface DisruptionPageInputs extends Partial<Disruption> {}
+export interface DisruptionPageInputs extends Partial<z.infer<typeof createDisruptionSchema>> {}
 
 const CreateDisruption = (initialState: PageState<Partial<DisruptionPageInputs>>): ReactElement => {
     const [pageState, setDisruptionPageState] = useState<PageState<Partial<DisruptionPageInputs>>>(initialState);
+    const [validityButtonClicked, setValidityButtonClicked] = useState(false);
```

Comment:
Seems like `validityButtonClicked` is used to show error in `TimeSelector` component. This is not required as it is not used in the  `TimeSelector` component